src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -1027,6 +1027,7 @@ private Result<?, List<UpdateError>> handleModifiedTrip(
                 .ifFailure(errors::add);
             }
           } else {
+            removePreviousRealtimeUpdate(trip, serviceDate);
```

Comment:
Should we move it outside the if check, as the same operation is done on both legs? Also should we call `cancelScheduledTrip` also on this one, which in turn calls `buffer.update`

Reply:
Seems logical Moved `removePreviousRealtimeUpdate` outside if statement and replaced `buffer.update` with `cancelScheduledTrip`.